### DISCUSSION TITLE
Getting started with waveshare's RP2040-ETH card.

### DISCUSSION THREAD

**Discussion by SamirKHE** [micropython/micropython#13237] [RP2040 / Pico]
Hello,
I got a waveshare RRP2040-ETH and i'm kinda lost as to how to start working with it so my questions are:
- official PICO firmeware seems to not have connectivity modules (like sockets) in it so which one should i use? 
- Any example on how to initiate a http request to a server with it? is there a driver for the CH9120 module?
Thanks in advance.

**Comment by ned-pcs** [comment]
I don't think there's a Micropython driver for the CH9120. There are some RP2 boards (like the Wiznet ones) that do have support for their Ethernet chips.

**Reply by jimmo** [reply]
That's right.

@SamirKHE the only wired-ethernet support currently available for the rp2040 is the WizNet-based chips.

**Comment by zpowellman** [comment]
The board contains a ch9120 serial/ethernet device.  It's basically a serial to ethernet converter.  It can operate in full duplex mode.  It can be configured as a tcp server or client or udp client or server.  It can be configured serially or by using the remote configuration app.  See the Waveshare rp2040-eth wiki for info and downloads.
https://www.waveshare.com/wiki/RP2040-ETH

**Reply by scruss** [reply]
Looks like this might include a library to configure and drive the CH9121 board: [Danielerikrust/CH9121: Software for the Waveshare Pico_ETH_CH9121](https://github.com/Danielerikrust/CH9121)

**Comment by nichokap** [comment]
You may have a look here [https://github.com/nichokap/RP2040-ETH](url)
